File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        for(string i : strs){
            string temp = i;
            sort(i.begin(), i.end());
            mp[i].push_back(temp);
        }
        vector<vector<string>> ret;
        ret.reserve(mp.size());
        for(pair i : mp){
            ret.push_back(i.second);
        }
        return ret;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = defaultdict(list)
        for string in strs:
            anagrams[''.join(sorted(string))].append(string)
        
        sol = []
        for _, value in anagrams.items():
            sol.append(value)
        
        return sol